crypto_analytics/collection/data_source/base.py

```diff
@@ -18,3 +21,34 @@ def write(self, filepath: str):
     @abstractmethod
     def get_time(self):
         pass
+
+
+class OHLCDataSource(DataSource):
+    """ An abstract class for all OHLC data sources """
+    def __init__(self, interval: Interval):
+        self.interval = interval
```

Comment:
rather than having interval as a parameter in the `fetch` method, I thought it would be better to set the interval as a property. My concern was that if we are trying to merge two data sources with different intervals, it would be useful to know that they have different intervals and so we would need a `interval` property to read from.